optimum/onnx/graph_transformations.py

```
@@ -79,3 +91,217 @@ def remove_duplicate_weights(model: ModelProto, inplace: bool = False) -> ModelP
+    """
+    _check_num_outputs(model1, model2)
+
+    for idx in range(len(model1.graph.output)):
+        model_output_1 = model1.graph.output[idx]
+        model_output_2 = model2.graph.output[idx]
+        if not model_output_1 == model_output_2:
+            if not (
+                model_output_1.name == model_output_2.name
+                and model_output_1.type.tensor_type.elem_type == model_output_2.type.tensor_type.elem_type
+            ):
+                raise ValueError(
+                    f"Can not match {model_output_1.name} with {model_output_2.name}. Make sure your"
+                    f" model protos have same outputs, have same data types and are in the same order."
+                )
+            model1.graph.output.remove(model_output_1)
+
+            new_output = onnx.helper.make_tensor_value_info(
+                model_output_2.name,
+                model_output_2.type.tensor_type.elem_type,
+                _infer_output_shape(model_output_2),
+            )
+            model1.graph.output.insert(idx, new_output)
+
+    if not all(
+        model_output_1 == model_output_2
+        for model_output_1, model_output_2 in zip(model1.graph.output, model2.graph.output)
+    ):
+        raise RuntimeError(f"Failed to unify outputs of given ONNX model protos.")
+
+
+def _get_all_inputs(model_list: List[ModelProto]):
+    inputs = []
+    input_names = []
+    for model in model_list:
+        for input in model.graph.input:
+            if input.name not in input_names:
+                input_names.append(input.name)
+                inputs.append(input)
+    return inputs
+
+
+def _get_onnx_opset(model: ModelProto):
+    opset_import = model.opset_import[0]
+    return getattr(opset_import, "version")
+
+
+def simplify_onnx_graph(model: ModelProto, print_info: bool = True) -> ModelProto:
+
+    model_simp, check = simplify(model)
+    assert check, "Simplified ONNX model could not be validated."
```

In general it is better to raise a proper exception than using asser (except in tests) IMO